.github/workflows/maven.yml

```diff
@@ -381,25 +400,48 @@ jobs:
           PGPASSWORD=password createdb -h localhost -U admin benchbase
       - name: Checkout repo
         uses: actions/checkout@v2
-      - name: Build docker image with all packages and run test suite
+      - name: Build benchbase-dev image
+        run: |
+          ./docker/benchbase/build-dev-image.sh
+      - name: Build the benchbase docker image with all profiles using the dev image
         env:
-          DOCKER_BUILDKIT: 1
-          BENCHBASE_PROFILES: 'cockroachdb mariadb mysql postgres spanner phoenix sqlserver'
+          SKIP_TESTS: 'false'
         run: |
-          docker build --build-arg "BENCHBASE_PROFILES=$BENCHBASE_PROFILES" --build-arg TEST_TARGET=test \
-            --build-arg CONTAINERUSER_UID=$UID --build-arg CONTAINERUSER_GID=$UID \
-            -t benchbase:latest -f ./docker/benchbase/Dockerfile --target fullimage .
-      - name: Run basic benchbase test from the docker image against postgres test DB
+          ./docker/benchbase/build-full-image.sh
+      - name: Run a basic test from the docker image against postgres test DB
         env:
             benchmark: noop
         run: |
-          # Adjust the sample config to talk to the container service instead of localhost.
-          cat "config/postgres/sample_${benchmark}_config.xml" | sed -e 's/localhost:5432/postgres:5432/g' > /tmp/config.xml
-          # Lookup the service container's docker network so we can place the benchbase container in it too.
-          docker_network="$(docker ps --filter expose=5432 --format '{{.Networks}}')"
-          # Map the adjusted config into the container and use it to run the test.
-          mkdir -p results
-          docker run --rm --name benchbase-postgres --network "$docker_network" \
-            --env BENCHBASE_PROFILE=postgres -v /tmp/config.xml:/tmp/config.xml -v "$PWD/results:/benchbase/results" \
-            benchbase:latest -b "$benchmark" -c /tmp/config.xml --create=true --load=true --execute=true
-          ls results/${benchmark}_*.csv
+          for image in benchbase benchbase-postgres; do
```

Comment:
Just a note for later:
One could refactor this a tad to build/test the singleton images as a parallel matrix strategy is done for the original build/testing workflows.
We already do that in a way that works so I opted for a smaller set of tests here for now.
One caveat for that if we wanted to change it in the future would be to separate out the shared `benchbase` image that builds all profiles into it so that the other jobs only push their own `benchbase-{profile}` image (as it currently stands those jobs would simultaneously produce a `benchbase` image that only has the single profile in it.

Reply:
I added a tweak for that too actually: 72aa4126